Add spreadsheet localization recipe (#28)
AckeeCZ/frontend-cookbook: pages/project/SpreadsheetTranslations.md

# 🇨🇿 Setup spreadsheet translations

There is a tool called [ackee-localize-spreadsheet-sdk](https://www.npmjs.com/package/ackee-localize-spreadsheet-sdk) that is built up to help us manage localizations in our applicationss that use `react-intl` library.

Our motivation for building such tool was a move of responsibility for filling translations out of repositories. We likely want to let customer (or some translations specialized department) fill translations independently on our source code.

> Here is few steps you need to go through to setup project for translations from spreadsheet

Reviewer: ```suggestion
Here are a few steps you need to go through to set up a project for translations from a spreadsheet
```

Author: never sure if it's according to the "few" or "steps"

* [Make a spreadsheet](#make-a-spreadsheet)
* [Setup config](#setup-config)
* [Install npm package](#install-npm-package)
* [Generate translations](#generate-translations)

## Make a spreadsheet

> Create a spreadsheet on Google drive. Better than make a clear spreadsheet is to duplicate one of existing, you can try [Abibuch project spreadsheet](https://docs.google.com/spreadsheets/d/1Klsl7XqBxPdeAgGA84kho49-_dwOpFXmWOeRzV3pWd4/edit#gid=0) or a [Spreadsheet template](https://docs.google.com/spreadsheets/d/1HKjvejcuHIY73WvEkipD7_dmF9dFeNLji3nS2RXcIzk/edit#gid=0).  

Reviewer: ```suggestion
Create a spreadsheet on Google drive. It is better to duplicate an existing spreadsheet rather than making a new one. You can try [Abibuch project spreadsheet](https://docs.google.com/spreadsheets/d/1Klsl7XqBxPdeAgGA84kho49-_dwOpFXmWOeRzV3pWd4/edit#gid=0) or a [Spreadsheet template](https://docs.google.com/spreadsheets/d/1HKjvejcuHIY73WvEkipD7_dmF9dFeNLji3nS2RXcIzk/edit#gid=0).  
```

**Don't forget to clone a spreadsheet to you own location.** The best place to keep project's spreadsheet is the `Projekty > My Project > Web` folder on Google Drive, where `My Project` is name of your project, eg. Abibuch. 

Important step is to **make the spreadsheet published**. Go to `File > Publishing on web` in your spreadshet and press `Publish` if there is one.

> 

Reviewer: ```suggestion

> Note: If you publish the whole spreadsheet you can use more tabs, not just one.

```

Reviewer: Tohle bych tam přidal, může se hodit vědět.

## Setup config

In your project root open `.ackeeconfig.json` (make one if it's not there) and fill these fields: `sheet_id`, `dir`, `cols`.

* `sheet_id` - id of your spreadsheet, you can get it from the spreadsheet url
* `dir` - directory to generate translation files into
* `cols` - comman separated list of languages to use. Names should correspond to column headers in the spreadsheet. Also they are used as a names for generated translation files.

Typical `.ackkeconfig.json` looks like
```json
{
    "localization": {
        "sheet_id": "1Klsl7XqBxPdeAgGA84kho49-_dwOpFXmWOeRzV3pWd4",
        "dir": "src/app/localization",
        "cols": "en,de",
        "type": "key_web"
    }
}
```

## Install npm package

Install [ackee-localize-spreadsheet-sdk](https://www.npmjs.com/package/ackee-localize-spreadsheet-sdk) to your project

```bash

> npm install ackee-localize-spreadsheet-sdk 

Reviewer: ```suggestion
npm install --save-dev ackee-localize-spreadsheet-sdk 
```
